src/common/components/feed/feed.utils.ts

```diff
@@ -134,19 +143,83 @@ export function generateFeedCardAttributes(
   }
 }
 
-export function shouldAggregate(e1: FeedQueryEvent, e2: FeedQueryEvent, e3: FeedQueryEvent) {
+export const getAggregateProfiles = (aggregateEvent: AggregateEvent): User[] => {
+  if (aggregateEvent.eventsAggregated.length < 2) {
+    return [
+      {
+        address: aggregateEvent.walletAddress,
+        profile: aggregateEvent.profile,
+      },
+    ];
+  }
+  const users: User[] = [];
+  aggregateEvent.eventsAggregated.map((user) => {
+    if (!users?.find((u) => u?.address === user.walletAddress)) {
+      users.push({
+        address: user.walletAddress,
+        profile: user.profile,
+      });
+    } else {
+      // do nothing
+    }
+  });
+  return users as User[];
+};
+
+export const aggregateEventsTime = (events: FeedQueryEvent[]) => {
+  let avgTime = 0;
+  events.forEach((event) => {
+    const date = DateTime.fromISO(event.createdAt).toMillis();
+    avgTime += date;
+  });
+  avgTime = avgTime / events.length;
+  const avgDate = DateTime.fromMillis(avgTime);
+  return avgDate;
+};
+
+export function shouldAggregateFollows(e1: FeedQueryEvent, e2: FeedQueryEvent, e3: FeedQueryEvent) {
```

Comment:
It's not clear to me why this takes 3 events rather than 2, or 5, or N. What are the consequences of passing in an arbitrarily sized array of events?

Reply:
should only aggregate after 3

Reply:
but yes I think thats a good change to be able to accept n